### ISSUE TITLE
Cannot read property 'start' of null" prevents request

### ISSUE THREAD

**Issue by mschipperheyn** [newrelic/newrelic-node-apollo-server-plugin#59]
## Description

I'm getting this error. It doesn't output in the server logs, but does show up on the client. The error completely prevents the request from processing. 

newrelic.js
```
exports.config = {
   agent_enabled: true,
   ssl: false,
   app_name: ['FNB Server'],
   license_key: 'XYZ',
   distributed_tracing: {
	enabled: false,
   },
   logging: {
     level: 'info'
   },
   allow_all_headers: true,
   attributes: {
      		exclude: [
			'request.headers.cookie',
			'request.headers.authorization',
			'request.headers.proxyAuthorization',
			'request.headers.setCookie*',
			'request.headers.x*',
			'response.headers.cookie',
			'response.headers.authorization',
			'response.headers.proxyAuthorization',
			'response.headers.setCookie*',
			'response.headers.x*',
		],
   }
}
```
## Expected Behavior

- In case of process errors, it should not prevent the request from completing
- Errors in processing should be logged.

## Your Environment

[TIP]:  # ( Include as many relevant details about your environment as possible. )

* Chrome
* newrelic: 6.14.0
* @newrelic/apollo-server-plugin: 0.1.1
* apollo-server-express: 2.18.2
* Node version: 12.18.2
* Mac OS

**Comment by nijotz** [comment]
Hi Marc, would you mind pasting the full stacktrace that you received to help us debug?

**Comment by michaelgoin** [comment]
Doing a quick scan, I'm guessing this is with the `segment.start()` calls? Seems like maybe there's some transaction state loss (which may want to work through via support ticket).

That being said, you are right in that we don't want the chance to throw an error and error out the GraphQL request.

**Comment by mschipperheyn** [comment]
Hi guys, let me see here. 
The stack is:
```
TypeError: Cannot read property 'start' of null"
    at Object.requestDidStart (/Users/mschipperheyn/projects/ology-rnw/packages/server/Users/mschipperheyn/projects/ology-rnw/node_modules/@newrelic/apollo-server-plugin/lib/create-plugin.js:82:1)"
    at initializeRequestListenerDispatcher (/Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/requestPipeline.js:305:1)"
    at Object.<anonymous> (/Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/requestPipeline.js:40:1)"
    at Generator.next (<anonymous>)"
    at /Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/requestPipeline.js:8:1"
    at new Promise (<anonymous>)"
    at module.exports.../../node_modules/apollo-server-core/dist/requestPipeline.js.__awaiter (/Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/requestPipeline.js:4:1)"
    at Object.processGraphQLRequest (/Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/requestPipeline.js:34:1)"
    at /Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/runHttpQuery.js:150:1"
    at Generator.next (<anonymous>)"
    at /Users/mschipperheyn/projects/ology-rnw/packages/server/build/webpack:/Users/mschipperheyn/projects/ology-rnw/node_modules/apollo-server-core/dist/runHttpQuery.js:8:1"
    at new Promise (<anonymous>)"
```

**Comment by michaelgoin** [comment]
@mschipperheyn thank you for the additional info!

**Comment by michaelgoin** [comment]
@mschipperheyn The crash part of the issue should be fixed in `v0.1.2` which was just shipped.

We'll still have to figure out what is causing the state loss.  

If you haven't already, the typical best next best step is to visit New Relic Support. As a part of that, we'll typically collect some logs (usually contain sensitive info), what modules you are using, etc. The typical cause is the introduction of a framework that adds an async boundary in a way that confuses the agent (or batches things up in a certain way). In this case, perhaps a middleware used with express?

It is also possible the additional plugins could do something that is not interacting well. Nothing obvious stands out on quick scan but worth experimenting with.
